### PR TITLE
Generate docs using Typedoc

### PR DESCRIPTION
**Pull request by jgroth**
This PR replaces Docz with Typedoc (which we were already kind of using :smile:)

fix Lundalogik/crm-feature#2773

### CODE REVIEW

File: etc/lime-web-components.api.md [hunk 1]
```diff
@@ -690,7 +690,9 @@ export interface Session {
     workstation: string;
 }
 
-// @public (undocumented)
+// Warning: (ae-internal-missing-underscore) The name "StateDecoratorConfig" should be prefixed with an underscore because the declaration is marked as @internal
```
[the remark this answers is not in the document]
Comment (jgroth) [reply etc/lime-web-components.api.md:693]: Yeah I was a bit annoyed by it... Maybe it's possible to turn it off somehow

**Comment by adrianschmidt** [comment]
Looks nice! Is the docs served when I run `npm start`? The cli output doesn't mention anything about it… 😄

**Comment by jgroth** [comment]
> Looks nice! Is the docs served when I run `npm start`? The cli output doesn't mention anything about it… smile

No, it will just run a watcher to build the interfaces. You can run `npm run docs -- --watch` though if you want a watcher for the docs

**Comment by adrianschmidt** [comment]
> You can run `npm run docs -- --watch` though if you want a watcher for the docs

But I take it we have nothing that will serve the resulting docs so you can look at them locally? I had a look at them by running `npx serve` in the `doc` folder, but I'm thinking it might be worth adding something simple to the npm scripts? Just adding `serve` as a dev-dependency and adding a separate `npm run serve` would be good enough for me. And maybe add a line about it in the readme.

**Comment by jgroth** [comment]
> > You can run `npm run docs -- --watch` though if you want a watcher for the docs
> 
> But I take it we have nothing that will serve the resulting docs so you can look at them locally? I had a look at them by running `npx serve` in the `doc` folder, but I'm thinking it might be worth adding something simple to the npm scripts? Just adding `serve` as a dev-dependency and adding a separate `npm run serve` would be good enough for me. And maybe add a line about it in the readme.

You can just open the `index.html` file in the browser :smile:

**Comment by jgroth** [comment]
But we can add something to serve them if you prefer that :slightly_smiling_face:

**Comment by jgroth** [comment]
Pulling this in to the cycle since it will fix Lundalogik/crm-feature#2773 :slightly_smiling_face:

**Comment by adrianschmidt** [comment]
> But we can add something to serve them if you prefer that 🙂

Nah… maybe if we feel we want hot reloading or something at some point. For now opening the file is good enough. But perhaps we can at least mention that in the repo's main readme.md?

**Review by adrianschmidt — APPROVED** [review]

**Pull request merged**